src/Phinx/Console/Command/Create.php

```diff
@@ -106,15 +123,57 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $fileName = Util::mapClassNameToFileName($className);
         $filePath = $path . DIRECTORY_SEPARATOR . $fileName;
 
-        if (file_exists($filePath)) {
+        if (is_file($filePath)) {
             throw new \InvalidArgumentException(sprintf(
                 'The file "%s" already exists',
                 $filePath
             ));
         }
 
-        // load the migration template
-        $contents = $this->getMigrationTemplate();
+        // Get the alternative template and static class options, but only allow one of them.
+        $altTemplate = $input->getOption('template');
+        $creationClassName = $input->getOption('class');
+        if ($altTemplate && $creationClassName) {
+            throw new \InvalidArgumentException('Cannot use --template and --class');
+        }
+
+        // Verify the alternative template file's existence.
+        if (!empty($altTemplate) && !file_exists($altTemplate)) {
+            throw new \InvalidArgumentException(sprintf(
+                'The alternative template file "%s" does not exist',
+                $altTemplate
+            ));
+        }
+
+        // Verify the static class exists and that it implements the required interface.
+        if (!empty($creationClassName)) {
+            if (!class_exists($creationClassName)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'The class "%s" does not exist',
+                    $creationClassName
+                ));
+            }
+            if (!is_subclass_of($creationClassName, self::CREATION_INTERFACE)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'The class "%s" does not implement the required interface "%s"',
+                    $creationClassName,
+                    self::CREATION_INTERFACE
+                ));
+            }
+        }
+
+        // Determine the appropriate mechanism to get the template
+        if (!empty($altTemplate)) {
+            // Get the template from the alternative template filename.
+            $contents = file_get_contents($altTemplate);
+        } elseif (!empty($creationClassName)) {
+            // Get the template from the creation class
+            $creationClass = new $creationClassName();
+            $contents = $creationClass->getMigrationTemplate();
+        } else {
+            // load the migration template
+            $contents = $this->getMigrationTemplate();
```

Comment:
I meant this call to `getMigrationTemplate`, which should really just return a path. If someone wants a synthetic template, instead of a file based on, they could use `CreationClass`.